google/cloud/logging_v2/handlers/_helpers.py

```diff
@@ -79,25 +80,29 @@ def get_request_data_from_flask():
     }
 
     # find trace id and span id
-    header = flask.request.headers.get(_FLASK_TRACE_HEADER)
-    trace_id, span_id = _parse_trace_span(header)
+    header = flask.request.headers.get(_FLASK_TRACEPARENT)
+    if header:
+        trace_id, span_id, trace_sampled = _parse_trace_parent(header)
```

Comment:
How do you suggest we let the caller know about it? I don't think we should throw an exception. We could print custom logs when we find unparseable metadata, but that's not something we've really done so far in the logging library code base (as far as I know)

Reply:
What you could do then to make best effort - try to call `_parse_trace_parent` and if it fails, call `_parse_xcloud_trace`. Also if you have a way to document somehow that it is best effort, it would be a plus

Reply:
Hmm ok, I changed the logic so it'll still work if both headers are set. It'll prefer `traceparent`, but if `traceparent` is invalid, it'll fall back to `xcloud_trace_context`.

I think the docs should make it clear all metadata is collected on a best-effort basis. I'll take a look to make sure when I update the usage guide for v3.0.0